### ISSUE TITLE
Why do we compute KL loss in this way？

### ISSUE THREAD

**Issue by jszgz** [hanzhanggit/StackGAN-v2#30]
def KL_loss(mu, logvar):
    # -0.5 * sum(1 + log(sigma^2) - mu^2 - sigma^2)
    KLD_element = mu.pow(2).add_(logvar.exp()).mul_(-1).add_(1).add_(logvar)
    KLD = torch.mean(KLD_element).mul_(-0.5)
    return KLD

**Comment by jszgz** [comment]
What does this formula mean？
